tests/integration/test_dirbrowser.py

```diff
@@ -0,0 +1,184 @@
+        return os.path.normpath(str(self.layout.join(*parts)))
+
+    def base_path(self):
+        """Return the path of the base temporary folder."""
+        return os.path.normpath(str(self.base))
+
+    def parse(self, quteproc):
+        """Parse the dirbrowser content from the given quteproc.
+
+        Args:
+            quteproc: The quteproc fixture.
+        """
+        html = quteproc.get_content(plain=False)
+        soup = bs4.BeautifulSoup(html, 'html.parser')
+        print(soup.prettify())
+        title_prefix = 'Browse directory: '
+        # Strip off the title prefix to obtain the path of the folder that
+        # we're browsing
+        path = soup.title.string[len(title_prefix):]
+        path = os.path.normpath(path)
+
+        container = soup('div', id='dirbrowserContainer')[0]
+
+        parent_elem = container('ul', class_='parent')
+        if len(parent_elem) == 0:
+            parent = None
+        else:
+            parent = QUrl(parent_elem[0].li.a['href']).toLocalFile()
+            parent = os.path.normpath(parent)
+
+        folders = []
+        files = []
+
+        for css_class, list_ in [('folders', folders), ('files', files)]:
+            for li in container('ul', class_=css_class)[0]('li'):
+                item_path = QUrl(li.a['href']).toLocalFile()
+                item_path = os.path.normpath(item_path)
+                list_.append(self.Item(path=item_path, link=li.a['href'],
+                                       text=str(li.a.string)))
+
+        return self.Parsed(path=path, parent=parent, folders=folders,
+                           files=files)
+
+
+@pytest.fixture(scope='module')
+def dir_layout(tmpdir_factory):
+    return DirLayout(tmpdir_factory)
+
+
+def test_parent_folder(dir_layout, quteproc):
+    quteproc.open_url(dir_layout.file_url())
+    page = dir_layout.parse(quteproc)
+    assert page.parent == dir_layout.base_path()
+
+
+def test_parent_with_slash(dir_layout, quteproc):
+    """Test the parent link with an URL that has a trailing slash."""
+    quteproc.open_url(dir_layout.file_url() + '/')
+    page = dir_layout.parse(quteproc)
+    assert page.parent == dir_layout.base_path()
+
+
+def test_enter_folder_smoke(dir_layout, quteproc):
+    quteproc.open_url(dir_layout.file_url())
+    quteproc.send_cmd(':hint all normal')
+    # a is the parent link, s is the first listed folder/file
+    quteproc.send_cmd(':follow-hint s')
+    page = dir_layout.parse(quteproc)
+    assert page.path == dir_layout.path('folder0')
+
+
+@pytest.mark.parametrize('folder', DirLayout.layout_folders())
+def test_enter_folder(dir_layout, quteproc, folder):
+    quteproc.open_url(dir_layout.file_url())
+    # Use Javascript and XPath to click the link that has the folder name as
+    # text.
+    quteproc.send_cmd(
+        ':jseval document.evaluate(\'//a[text()="{}"]\', document, null, '
+        'XPathResult.ANY_TYPE, null).iterateNext().click()'
+        .format(folder)
+    )
```

Comment:
Hmm, maybe that should be a `quteproc.click_element(text=...)` method or so? Do you think it could be useful for other tests too?